Tests/DataCollector/ContentfulDataCollectorTest.php

```diff
@@ -32,4 +34,32 @@ public function testGetClients()
 
         $this->assertEquals($clients, $dataCollector->getClients());
     }
+
+    public function testReset()
+    {
+        $logEntry = $this->prophesize(LogEntry::class);
+        $expectedLogs = [$logEntry->reveal()];
+
+        /** @var ArrayLogger|ObjectProphecy $logger */
+        $logger = $this->prophesize(ArrayLogger::class);
+        $logger
+            ->getLogs()
+            ->willReturn($expectedLogs);
+
+        /** @var Request|ObjectProphecy $request */
+        $request = $this->prophesize(Request::class);
+        /** @var Response|ObjectProphecy $response */
+        $response = $this->prophesize(Response::class);
+
+        // initialize collector
+        $collector = new ContentfulDataCollector($logger->reveal());
+
+        // check whether collector retrieves log entries from logger
+        $collector->collect($request->reveal(), $response->reveal());
+        $this->assertSame($expectedLogs, $collector->getLogs());
+
+        // check whether collectors log entries are empty after resetting
+        $collector->reset();
+        $this->assertSame([], $collector->getLogs());
```

Comment:
Would it be possible to change this and not use Prophecy, at least for Request and Response? I think they're both rather easy to just use, they can just be created as `new Request() / new Response()` (plus, `$collector->collect()` does not even really use them.

I'll have to change the LogEntry/ArrayLogger part myself because CDA SDK v3 changed the way that that works, though.